Source/StorageUtils.swift

```diff
@@ -64,7 +79,25 @@ class StorageUtils {
     }
     
     static func iabGdprConsent() -> String? {
-        return getObjectFromUserDefaults(forKey: StorageUtils.IABConsent_ConsentStringKey)
+        var gdprConsentString:String? = getObjectFromUserDefaults(forKey: StorageUtils.IABTCF_ConsentString)!
+        
+        if(gdprConsentString == nil || gdprConsentString == String.EMPTY_String){
+            gdprConsentString = getObjectFromUserDefaults(forKey: StorageUtils.IABConsent_ConsentStringKey)!
+        }
+        return gdprConsentString
+        
+    }
+    
+    static func setPurposeConsents(value: String?) {
```

Comment:
maybe we can use Bool instead of String ?

Reply:
purposeConsent is a string like this "1010101001" from this only the 1st digit is the PurposeConsent all other are for different consents provided

Reply:
Where can I found this information ? I mean where is described that the first char is a purposeConsent ?

Reply:
https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20Consent%20string%20and%20vendor%20list%20formats%20v2.md